sdk/iot/provisioning/src/az_iot_provisioning_client.c

```diff
@@ -10,3 +10,175 @@
 
 #include <_az_cfg.h>
 
+static const az_span provisioning_service_api_version
+    = AZ_SPAN_LITERAL_FROM_STR("/api-version=" AZ_IOT_PROVISIONING_SERVICE_VERSION);
+
+static const az_span str_dps = AZ_SPAN_LITERAL_FROM_STR("$dps");
+static const az_span str_registrations = AZ_SPAN_LITERAL_FROM_STR("/registrations/");
+static const az_span str_put_iotdps_register
+    = AZ_SPAN_LITERAL_FROM_STR("PUT/iotdps-register/?$rid=1");
+static const az_span str_get_iotdps_get_operationstatus
+    = AZ_SPAN_LITERAL_FROM_STR("GET/iotdps-get-operationstatus/?$rid=1&operationId=");
+static const az_span str_res = AZ_SPAN_LITERAL_FROM_STR("res/");
+
+AZ_NODISCARD az_iot_provisioning_client_options az_iot_provisioning_client_options_default()
+{
+  return (az_iot_provisioning_client_options){ .user_agent = AZ_SPAN_NULL };
+}
+
+AZ_NODISCARD az_result az_iot_provisioning_client_init(
+    az_iot_provisioning_client* client,
+    az_span global_device_endpoint,
```

Comment:
>global_device_endpoint [](start = 12, length = 22)

can this be called endpoint?

Reply:
I don't recommend it: Device Provisioning has _both_ a service client endpoint (which is particular to each instance) and the global endpoint which is common to _all_ instances in a particular cloud.
In general, gde will be the same for most of our customers unless they are part of one of the non-standard clouds.

Reply:
Are we going to expose one more init for Service endpoint?

---
In reply to: [404454304](https://github.com/Azure/azure-sdk-for-c/pull/547#discussion_r404454304) [](ancestors = 404454304)